memory_leak_worker_logging/evm_mem_log_processor.rb

```diff
@@ -37,27 +39,18 @@
 worker_type_regexp += options[:worker_type] if options[:worker_type]
 
 LOG_LINE_REGEXP = Regexp.new [
-  /^\[----\] ([A-Z]), /,               # 1. severity char
-  /\[([-0-9]*)/,                       # 2. date
-  /T([0-9:]*)/,                        # 3. Ttime
-  /\.([0-9]*) /,                       # 4. .ms
-  /#([0-9]*)/,                         # 5. #pid
-  /:([a-z0-9]*)\] +/,                  # 6. :thread
+  /(?<SEVERITY_CHAR>[A-Z]), \[/,       # 1. severity char
+  /(?<date>[-0-9]*)T/,                 # 2. date
+  /(?<TIME>[0-9:]*\.)/,                # 3. Ttime
+  /(?<MS>[0-9]*) #/,                   # 4. .ms
+  /(?<PID>[0-9]*):/,                   # 5. #pid
+  /(?<THREAD>[a-z0-9]*)\] +/,          # 6. :thread
   /[A-Z]{0,5} -- [^:]*: /,
   /MIQ\(#{worker_type_regexp}\) .*/,
   /Memory Info XXXX /,
-  /(\([^\)]*\) )?=> /,                 # 7. msg (queue msg, etc)
-  /(\{[^\}]*\})/                       # 8. mem info
-].map(&:source).join
-
-DATE=2
-TIME=3
-PID=5
-MSG=7
-INFO=8
-
-data_file   = nil
-current_pid = nil
+  /(?<MSG>\([^\)]*\) )?=> /,           # 7. msg (queue msg, etc)
+  /(?<MEM_INFO>\{[^\}]*\})/            # 8. mem info
+].map(&:source).unshift(/^\[----\] /.source).join
```

Comment:
Yeah... probably could... not sure what I was thinking with this one...

That said, since I opened up #8, I will probably reusing this first part and dumping it in the `util/` folder, since it will be useful in other parsers as well.

Thanks for calling me out on this one, it was dumb.